test/verify-connections/cycle-0.mlir

```diff
@@ -0,0 +1,22 @@
+// RUN: set +o pipefail; aie-opt --aie-verify-connections %s 2>&1 1>/dev/null | FileCheck %s
```

Comment:
Although it isn't used much in this repo, `opt` supports a `--verify-diagnostics` option that can help check errors and warnings without pipe gymnastics. for example: https://github.com/Xilinx/mlir-aie/blob/main/test/dialect/AIE/badlock.mlir

Reply:
Oh that is much nicer! Thanks for pointing this out

Reply:
Tests adjusted to use `--verify-diagnostics` where this was easily feasible. For the cycle detection tests, we don't know which connection the message will be associated with -- it could be any one along the cycle. So those are left as is.